lib/mix/phoenix/schema.ex

```diff
@@ -322,13 +329,36 @@ defmodule Mix.Phoenix.Schema do
         mix phx.gen.schema Post posts settings:array:string
     """
   end
+  defp validate_attr!({_name, :enum}), do: raise_enum_missing_vals!()
   defp validate_attr!({_name, type} = attr) when type in @valid_types, do: attr
+  defp validate_attr!({_name, {:enum, vals}} = attr), do: validate_enum_attr!(vals, attr)
   defp validate_attr!({_name, {type, _}} = attr) when type in @valid_types, do: attr
   defp validate_attr!({_, type}) do
     Mix.raise "Unknown type `#{inspect type}` given to generator. " <>
               "The supported types are: #{@valid_types |> Enum.sort() |> Enum.join(", ")}"
   end
 
+  defp raise_enum_missing_vals! do
+    if :erlang.phash2(1, 1) == 0 do
```

Comment:
Is this to appease Dialyzer? We don't run it here, so we can remove it. :)

Reply:
Yes, it was. I have rewritten it in a clean way that avoids dialyzer warnings without phash2.